controlplane/kubeadm/api/v1beta1/kubeadmcontrolplanetemplate_types.go

```diff
@@ -53,5 +56,72 @@ func init() {
 
 // KubeadmControlPlaneTemplateResource describes the data needed to create a KubeadmControlPlane from a template.
 type KubeadmControlPlaneTemplateResource struct {
-	Spec KubeadmControlPlaneSpec `json:"spec"`
+	Spec KubeadmControlPlaneTemplateResourceSpec `json:"spec"`
+}
+
+// KubeadmControlPlaneTemplateResourceSpec defines the desired state of KubeadmControlPlane.
```

Comment:
Context on why we want to drop the fields in: https://github.com/kubernetes-sigs/cluster-api/issues/5678

I think we have 3 options:

### 1. Keep the fields for the lifetime of KCP/CAPI

This forces users to specify fields which are not used, which is a long-term UX problem imho.

### 2. Break Go (struct) compatibility, deprecate the fields, drop them with v1beta2

This would allow users to just not set the values which are not needed / used. Existing YAMLs still work. And we're able to drop the fields with v1beta2.

### 3. Break Go (struct) and YAML compatibility, drop the fields now

This would allow us to drop the fields as soon as possible, but this will break YAMLs which contain those fields.

Reply:
In my opinion we shouldn't keep the fields as the impact of keeping them is far worse then the breaking change.

I personally lean towards option 2 as it allows for a smooth migration at least for YAMLs, and probably almost nobody is using the KCPTemplate struct right now, and even if they are then can just not set the fields anymore, which weren't used anyway.

Reply:
cc @vincepri @fabriziopandini 
 
 Opinions?
 I think in any case we should bring this topic up on the CAPI meeting next week.

Reply:
Just talked to Fabrizio and Vince and we would go with Option 3, given that the fields are currently not used at all. So if users migrate to CAPI v1.1 they can just drop those fields without losing anything.

@enxebre @CecileRobertMichon WDYT?

Reply:
+1 on dropping the fields now

Reply:
+1